src/inc-lib-init.php

```diff
@@ -1113,28 +1113,53 @@ function lovd_getVariantInfo ($sVariant, $sTranscriptID = '', $bCheckHGVS = fals
     // Trim the variant and remove whitespaces.
     $sVariant = preg_replace('/\s+/', '', $sVariant);
 
-    if (preg_match('/^[NX][MR]_[0-9]{6,9}\.[0-9]+:[cn]/', $sVariant)) {
-        if (is_numeric($sTranscriptID)) {
-            $sNCBIID = $_DB->query('SELECT id_ncbi FROM ' . TABLE_TRANSCRIPTS . ' WHERE id = ?',
-                array($sTranscriptID)) ->fetchColumn();
-        } else {
-            $sNCBIID = $sTranscriptID;
-        }
 
-        if ($sNCBIID == strstr($sVariant, ':', true)) {
-            // The transcript given in the DNA description is also
-            //  the transcript that we're using in LOVD for this variant.
-            $aResponse['warnings']['WTRANSCRIPTFOUND'] = 'Transcript ID found in the DNA description.';
-        } else {
-            // This is an actual problem; the submitter used a
-            //  different transcript than configured in LOVD.
-            $aResponse['warnings']['WDIFFERENTTRANSCRIPT'] =
-                'The transcript found in the DNA description does not match the configured transcript.';
+    // Match the reference sequence if one was given.
+    if (preg_match('/^(' .
+        '(\(?[NX][CGMRTW]_[0-9]{6,9}.[0-9]+\)?){1,2}|' .
+        'ENS[TG][0-9]{11}.[0-9]+|' .
+        'LRG_[0-9]{3}(t[0-9]+)?' .
+        ')/', $sVariant)) {
+        // We are checking to see whether the variant sstarts with
+        //  a reference sequence such as NC_123456.1:...
+        //  or ENST12345678911.1:...
+        if ($sTranscriptID !== false) {
+            if (is_numeric($sTranscriptID)) {
+                $sNCBIID = $_DB->query('SELECT id_ncbi FROM ' . TABLE_TRANSCRIPTS . ' WHERE id = ?',
+                    array($sTranscriptID))->fetchColumn();
+            } else {
+                $sNCBIID = $sTranscriptID;
+            }
+
+            if ($sNCBIID == strstr($sVariant, ':', true)) {
+                // The transcript given in the DNA description is also
+                //  the transcript that we're using in LOVD for this variant.
+                $aResponse['warnings']['WTRANSCRIPTFOUND'] = 'Transcript ID found in the DNA description.';
+            } else {
+                // This is an actual problem; the submitter used a
+                //  different transcript than configured in LOVD.
+                $aResponse['warnings']['WDIFFERENTTRANSCRIPT'] =
+                    'The transcript found in the DNA description does not match the configured transcript.';
```

Comment:
This is actually not true for NC(NM) notation. This code may need some work, but we can do that later.